src/error.rs

```diff
@@ -201,6 +200,11 @@ impl Error {
         self.find_source::<TimedOut>().is_some()
     }
 
+    /// Returns true if the error was caused by a shutdown.
+    pub fn is_shutdown(&self) -> bool {
```

Comment:
e.g.

```suggestion
    #[cfg(feature = "http1")]
    pub fn is_http1_shutdown(&self) -> bool {
```

Either that, or this should probably also handle h2 goaways?

Reply:
Hm, that does make me think the name could confuse people. It's not a graceful shutdown, it was an error that occurred when calling `AsyncWrite::shutdown`.

It originally was just meant to provide additional context on top of the IO error. Is that context useful on its own?

Reply:
I don't understand what you mean. There is nothing we can do about those errors in the wild (most often they are a "broken pipe" error), so we don't want to log them at all, so we need to discriminate them to ignore them.

Reply:
So, it does seem like the context is useful to you. Now the question is, is the name confusing, such that someone might think it's a "graceful shutdown" error received over HTTP/2? Is that a bad thing? Should we hijack the name to make both cases be a `is_shutdown` error?

Reply:
h2 errors currently are all represented as `Kind::H2` right? Maybe I should rename that method to `is_http1_shutdown` then.